### ISSUE TITLE
seq2seq model with transformer

### ISSUE THREAD

**Issue by juliahane** [huggingface/transformers#988]
Hi
I am urgently looking for a sequence to sequence model with transformer with script to
finetuning and training, I appreciate telling me which of the implementations in this repo could do a sequence to sequence model?
thanks
Best regards
Julia

**Comment by LysandreJik** [comment]
Are you looking for LSTM/RNN-based seq2seq architectures or Transformer-based architectures? This repository does not host any LSTM/RNN architectures.

You can find information on all our (transformer) [models here](https://huggingface.co/pytorch-transformers/pretrained_models.html), and [examples using them here](https://huggingface.co/pytorch-transformers/examples.html).

**Comment by juliahane** [comment]
I am looking for transformer based, pretrained model, I am not sure which of  the implemented models in this repo I can use for seq2seq model? thanks for your help

**Comment by LysandreJik** [comment]
The models hosted on this repo unfortunately probably cannot be used in a traditional sequence-to-sequence manner like translation (if that's what you have in mind).

**Comment by juliahane** [comment]
yes, exactly, I am looking for such models, even gpt model cannot be used
for this purpose?
or gpt2 by conditioning?
Are you aware of clean implementation for seq2seq model with any of these
pretrained models hosted in your repo?
thanks.

**Comment by julien-c** [comment]
Hi @juliahane, maybe take a look at `fairseq`

**Comment by juliahane** [comment]
Hi
Thanks, Do you mind also suggest me a good implementation with lstm for
seq2seq model, I need some implementation with high quality of decoding,
thanks.

**Comment by juliahane** [comment]
Hi
I found FairSeq implementation not really clean and modular code. Are you
aware of more work which extend BERT, GPT, ... to a language model with
decoder?
thanks
Julia

**Comment by stefan-it** [comment]
Then you should have a look at the "Cross-lingual Language Model Pretraining" from Lample and Conneau: https://arxiv.org/abs/1901.07291

Implementation of supervised and unsupervised NMT can be found here: https://github.com/facebookresearch/XLM#iii-applications-supervised--unsupervised-mt :)

**Comment by juliahane** [comment]
Hi
thanks a lot. I was wondering if you could also suggest me a good
implementation for seq2seq with LSTMs in pytorch with good accuracy.
I have a deadline and I cannot find any, I really appreciate your help.
thanks
Julia

**Comment by LysandreJik** [comment]
Hey Julia, without a specific task in mind I can't think of anything relevant, but browsing [paperswithcode.com with a seq2seq search](https://paperswithcode.com/search?q_meta=&q=seq2seq) yields quite a few interesting results.

**Comment by juliahane** [comment]
Hi
My task is a autoencoding text. So encoding and decoding it in one
language. Thanks

**Comment by juliahane** [comment]
I was wondering if you could tell me which of these are a fast sequence to
sequence implementation,
this is really hard for me to figure out which one to use. thanks

**Comment by juliahane** [comment]
I did checked this implementations you sent me, I honestly cannot find a
single good seq2seq one with lstm, and I really appreciate your help

**Comment by stefan-it** [comment]
@juliahane `fairseq` has an example of how to use a LSTM (encoder & decoder) for a seq2seq model:

https://fairseq.readthedocs.io/en/latest/tutorial_simple_lstm.html

Additionally, you could also check out Joey NMT, which has a very nice and clear codebase:

https://github.com/joeynmt/joeynmt

**Comment by juliahane** [comment]
Hi
Thanks, Fairseq to me is not following a good coding practice although
Facebook has published it,
but the second one looks much better, thank you.

I was wondering if you could tell me if torchtext is faster than using
dataloader in pytorch for seq2seq applications?
I wonder how torchtext impact the speed and if  this is really better than
dataloader

thanks

**Comment by thomwolf** [comment]
Merging with #1506
